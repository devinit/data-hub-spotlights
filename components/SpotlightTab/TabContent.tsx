import React, { FunctionComponent } from 'react';

const TabContent: FunctionComponent = ({ children }) => {
  return (
    <article className="tabs__content">
      { children }
    </article>
  );
};

export { TabContent };
