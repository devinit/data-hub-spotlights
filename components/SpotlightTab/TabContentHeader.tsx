import React, { FunctionComponent } from 'react';

const TabContentHeader: FunctionComponent = ({ children }) => {
  return (
    <div className="tabs__content__header">
      { children }
    </div>
  );
};

export { TabContentHeader };
