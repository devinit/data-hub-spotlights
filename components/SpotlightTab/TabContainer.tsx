import React, { Children, FunctionComponent, isValidElement } from 'react';
import { TabContent } from './TabContent';

interface TabContainerProps {
  id: string;
  label: string;
  active?: boolean;
}

const TabContainer: FunctionComponent<TabContainerProps> = ({ id, active, label, children }) => {
  const renderContent = () => {
    return Children.map(
      children,
      (child) => isValidElement(child) && child.type === TabContent ? child : null
    );
  };

  return (
    <section className="tabs__container" id={ id }>
      <input className="tabs__input" type="radio" name="sections" id={ `${id}-option` } checked={ active }/>
      <label className="tabs__label" htmlFor={ `${id}-option` }>{ label }</label>
      { renderContent() }
    </section>
  );
};

export { TabContainer };
