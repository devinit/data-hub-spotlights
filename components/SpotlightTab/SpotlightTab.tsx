import React, { Children, FunctionComponent, isValidElement } from 'react';
import { TabContainer } from './TabContainer';

const SpotlightTab: FunctionComponent = ({ children }) => {
  const renderTabs = () => {
    return Children.map(
      children,
      (child) => isValidElement(child) && child.type === TabContainer ? child : null
    );
  };

  return (
    <div className="tabs">
      { renderTabs() }
    </div>
  );
};

export { SpotlightTab };
