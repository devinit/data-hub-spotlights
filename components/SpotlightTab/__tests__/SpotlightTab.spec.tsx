/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightTab } from '../SpotlightTab';

describe('SpotlightTab', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightTab/>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
