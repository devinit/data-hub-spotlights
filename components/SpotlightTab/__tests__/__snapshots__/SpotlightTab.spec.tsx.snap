// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightTab renders correctly 1`] = `
<div
  className="tabs"
/>
`;
